src/rulesets/oas/index.json

```diff
@@ -115,7 +115,12 @@
       "description": "Example should have either a `value` or `externalValue` field.",
       "recommended": true,
       "type": "style",
-      "given": "$..examples.*",
+      "given": [
```

Comment:
@m-mohr This rule currently applies to both OAS2 and OAS3.

Could you please double check that all the new paths make sense in both context (cf. https://github.com/OAI/OpenAPI-Specification/blob/master/schemas/v2.0/schema.json and https://github.com/OAI/OpenAPI-Specification/blob/master/schemas/v3.0/schema.json).

Wouldn't that be the case, an approach would be to create two new rules (prefixed `oas2-` and `oas3-`) that would only target the relevant paths considering the spec version.

Reply:
@P0lip would dropping a rule and creating two new ones be considered as a breaking change?

Reply:
I'd rather avoid creating a new rule if we can help it, but if we *really* need to introduce a separate rule, that wouldn't necessarily be a breaking change.
Is there any way we can leverage multiple paths to target relevant parts of the document?

Reply:
@nulltoken Are you sure this applies to OAS2? I can't find any info on examples with value or externalValue in the OAS2 spec. 🤔

Reply:
@m-mohr Sorry, I was unclear. Let me try and reword my thoughts: The currently merged rule applies to both OAS2 and OAS3 (with the `"$..examples.*"` JsonPath expression).

I was requesting you to double check that all the new paths this PR adds also apply to OAS2 and OAS3. If that's not the case, then I fear we'll have to eventually have to end up with two rules: one for OAS2 and one for OAS3, each of them leveraging the `format` rule attribute as a target specifier.

Reply:
@m-mohr Duh. My brain just exploded for a minute. 😜 

You're right. 

Although the path applies to both specs, the rule only makes sense in OAS3 context (as hinted in #883).

How about about renaming that rule to `oas3-example-value-or-externalValue` and adding a `formats` specifier to make sure it only considers OAS3 documents?

Reply:
Sure, would make sense, but is a breaking change. So I'd probably prefer to merge this as non-breaking change to get the fix faster into a release and then change the name for the next major release.

Reply:
>So I'd probably prefer to merge this as non-breaking change to get the fix faster into a release

@m-mohr :+1: Let's deal with this later as part of #883.